swift/swift.go

```diff
@@ -188,3 +204,67 @@ func maybeNotFound(err error, format string, arg ...interface{}) error {
 	}
 	return errors.Newf(err, format, arg...)
 }
+
+type ohandler struct {
+	c             *Client
+	containerName string
+	objectName    string
+	length        int64
+	pos           int64
+}
+
+// Read reads up to len(b) bytes from the handler. It returns the number of bytes read and any error encountered. At end of stream, Read returns no EOF on the first EOF and EOF after that.
+func (o *ohandler) Read(p []byte) (n int, err error) {
+	if o.pos == o.length {
+		return 0, io.EOF
+	}
+	end := int64(len(p)) + o.pos
+	header := make(http.Header)
+	header.Add("Range", fmt.Sprintf("bytes=%d-%d", o.pos, end))
+	requestData := goosehttp.RequestData{
+		RespReader:     &emptyReadCloser,
+		ReqHeaders:     header,
+		ExpectedStatus: []int{http.StatusOK, http.StatusPartialContent},
+	}
+	err = o.c.touchObject(&requestData, client.GET, o.containerName, o.objectName)
+	if err != nil {
+		return
+	}
+	defer requestData.RespReader.Close()
+	readSize := 0
+	for n < len(p) {
+		readSize, err = requestData.RespReader.Read(p[n:])
+		n += readSize
+		o.pos += int64(readSize)
+		if err != nil {
+			if err == io.EOF || err == io.ErrUnexpectedEOF {
```

Comment:
This seems odd, whilst it's reasonable to treat EOF as not an error you are worried about, ErrUnexpectedEOF should probably never be ignored like this. If it is really the behavior you want then perhaps add a comment.